exporter/otlpexporter/otlp.go

```diff
@@ -192,6 +203,14 @@ func processError(err error) error {
 	return err
 }
 
+func (e *baseExporter) reportStatusFromError(err error) {
+	if err != nil {
+		componentstatus.ReportStatus(e.host, componentstatus.NewRecoverableErrorEvent(err))
+		return
+	}
+	componentstatus.ReportStatus(e.host, componentstatus.NewEvent(componentstatus.StatusOK))
```

Comment:
Are we not concerned for every request to report the status? Should we do it periodically, or only when changes?

Reply:
The [finite state machine](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-status.md#transitioning-between-statuses) handles this for us. It will report status only if there has been a change, otherwise it will no-op.